### PR TITLE
AYS-487 | `AysUserNotSuperAdminException` Has Been Extended from `AysForbiddenException` Instead of `AccessDeniedException` Class

### PR DESCRIPTION
**Pull request by suhakopan**
### Checklist

Before submitting your pull request, ensure the following:

- [x] **Title and Branch Naming Conventions:**
    - The pull request title follows the
      standard: [Pull Request Naming Conventions](https://github.com/afet-yonetim-sistemi/ays-be/blob/main/CONTRIBUTING.md#pull-request-naming-conventions).
    - The branch name follows one of the conventions outlined in
      the [Branch Naming Conventions](https://github.com/afet-yonetim-sistemi/ays-be/blob/main/CONTRIBUTING.md#branch-naming-conventions).

- [x] **Local Testing:**
    - I have tested my changes locally on Postman, and they are working as expected.

- [x] **Code Quality:**
    - The code is formatted according to the project's coding guidelines and style.
    - The code has been reviewed to ensure its quality.
    - The code does not contain any issues flagged by SonarLint.

- [x] **Documentation:**
    - Necessary documentation has been added or existing documentation has been updated, specifically detailing changes made in Postman.

- [x] **Testing:**
    - Relevant unit tests have been written and included.
    - Relevant integration tests have been written and included.

- [x] **Reviewers and Assignees:**
    - Default reviewers have been assigned to this pull request.
    - Assignees have been added if necessary.

- [x] **Labels and Associations:**
    - No specific actions are required in the Labels and Associations section for this pull request.

### CODE REVIEW

File: src/main/java/org/ays/common/util/exception/handler/GlobalExceptionHandler.java [hunk 1]
```diff
@@ -160,6 +160,7 @@ AysErrorResponse handleAccessDeniedError(final AccessDeniedException exception)
 
         return AysErrorResponse.builder()
                 .header(AysErrorResponse.Header.AUTH_ERROR.getName())
+                .message(exception.getMessage())
```
Comment (agitrubard) [inline src/main/java/org/ays/common/util/exception/handler/GlobalExceptionHandler.java:163]: @suhakopan burada Access Denied için message dönmektense yeni bir tane `AysForbiddenException` açıp asıl sınıfı oradan extend edelim. Bu sayede daha düzgün olur. Bir öncekinde hata aslında buradaydı.

**Review by agitrubard — CHANGES_REQUESTED** [review]

File: src/main/java/org/ays/common/util/exception/handler/GlobalExceptionHandler.java (hunk 1 above)
Comment (suhakopan) [reply src/main/java/org/ays/common/util/exception/handler/GlobalExceptionHandler.java:163]: Bunu kaldırıp problemin asıl yeri olan AysUserNotSuperAdminException için exception handler yazsak doğru olmaz mı?

Comment (agitrubard) [reply src/main/java/org/ays/common/util/exception/handler/GlobalExceptionHandler.java:163]: @suhakopan her exception'ı handle etmeye kalkarsak muhtemelen yönetemeyiz, bu yüzden base exception'ları belirleyip sadece extend etmek yeterli olur, çünkü ilerleyen dönemde başka exception'lar da gelirse onları da eklememiz gerekecek

**Comment by agitrubard** [comment]
@suhakopan ilgilenebilecek misin? Eğer fırsatın olmayacak ise ben bakabilirim. Bu konuda bilgi verebilirsen çok iyi olur :raise

**Review by agitrubard — APPROVED** [review]

**Pull request merged**